pydarn/plotting/fan.py

```diff
@@ -0,0 +1,248 @@
+        scan = np.zeros((fan_shape[0] - 1, fan_shape[1]-1))
+        grndsct = np.zeros((fan_shape[0] - 1, fan_shape[1]-1))
+        iterat = 0
+        for i in np.nditer(plot_beams):
+            try:
+                slist = dmap_data[i.astype(int)]['slist']
+                scan[slist, iterat] = dmap_data[i.astype(int)][parameter]
+                grndsct[slist, iterat] = dmap_data[i.astype(int)]['gflg']
+                iterat += 1
+            except KeyError:
+                continue
+
+        # Colour table and max value selection depending on parameter plotted
+        # Load defaults if none given
+        # TODO: use cmaps as over writting cmap is bad practice...
+        # did I do that in my code ... hmm
+        if cmap is None:
+            cmap = {'p_l': 'plasma', 'v': PyDARNColormaps.PYDARN_VELOCITY,
+                    'w_l': PyDARNColormaps.PYDARN_VIRIDIS,
+                    'elv': PyDARNColormaps.PYDARN}
+            cmap = plt.cm.get_cmap(cmap[parameter])
+
+        # Setting zmin and zmax
+        defaultzminmax = {'p_l': [0, 50], 'v': [-200, 200],
+                          'w_l': [0, 250], 'elv': [0, 50]}
+        if zmin is None:
+            zmin = defaultzminmax[parameter][0]
+        if zmax is None:
+            zmax = defaultzminmax[parameter][1]
+
+        # Setup plot
+        # TODO: eek! remeber there is an option to pass in a variable called ax
+        # This may screw up references
+        ax = plt.axes(polar=True)
+        if hemisphere == 'south':
+            ax.set_ylim(-90, -lowlat)
+            ax.set_yticks(np.arange(-lowlat, -90, -10))
+        else:
+            ax.set_ylim(90, lowlat)
+            ax.set_yticks(np.arange(lowlat, 90, 10))
+        ax.set_xticklabels(['00', '', '06', '', '12', '', '18', ''])
+        ax.set_theta_zero_location("S")
+
+        # For plotting all gates or just 75
+        if all_gates is True:
+            rangelimit = thetas.shape[0]
+        else:
+            rangelimit = 75
+
+        # Plotting! WOO!
+        for gates in range(rangelimit - 1):
+            for beams in range(thetas.shape[1] - 2):
+                # Index colour table correctly
+                cmapindex = (scan[gates, beams] + abs(zmin)) /\
+                        (abs(zmin) + abs(zmax))
+                if cmapindex < 0:
+                    cmapindex = 0
+
+                if cmapindex > 1:
+                    cmapindex = 1
+                colour_rgba = cmap(cmapindex)
+
+                # Check for zero values (white) and groundscatter (gray)
+                if scan[gates, beams] == 0:
+                    colour_rgba = 'w'
+
+                if groundscatter and grndsct[gates, beams] == 1:
+                    colour_rgba = 'gray'
+
+                theta = [thetas[gates, beams], thetas[gates + 1, beams],
+                         thetas[gates + 1, beams + 1],
+                         thetas[gates, beams + 1]]
+                # TODO: try to avoid single letter names or at
```

Comment:
review this TODO please :)

Reply:
I would say it's common to use `r` and `theta` as polar plot variables. Added comments